addons/io_scene_gltf/node.py

```diff
@@ -181,26 +175,75 @@ def add_bone(idx, parent, parent_mat):
     bpy.context.scene.objects.unlink(arma_ob)
 
 
-def create_scene(op, idx):
+def create_node(op, idx):
+    node = op.gltf['nodes'][idx]
+
+    # print("Creating node: {}".format(idx) )
+
+    def create(name, data):
+        ob = bpy.data.objects.new(name, data)
+        bpy.context.scene.objects.link(ob)
+        return ob
+
+    objects = []
+    if 'mesh' in node:
+        mesh_name = node.get('name', 'mesh[%d]' % idx)
+        mesh = create(mesh_name, op.get_mesh(node['mesh']))
+        objects.append(mesh)
+
+    if 'camera' in node:
+        camera_name = node.get('name', 'camera[%d]' % idx)
+        camera = create(camera_name, op.get_camera(node['camera']))
+        objects.append(camera)
+
+    if not objects:
+        name = node.get('name', 'node[%d]' % idx)
+        objects.append(create(name, None))
+
+    for obj in objects:
+        set_transform(node, obj)
+
+    parent = objects[0]
+    children = node.get('children', [])
+    for child_idx in children:
+        for child_node in create_node(op, child_idx):
+            child_node.parent = parent
+
+    return objects
+
+
+def create_scene(op, idx, blender_scene=None):
     scene = op.gltf['scenes'][idx]
     name = scene.get('name', 'scene[%d]' % idx)
 
-    bpy.ops.scene.new(type='NEW')
-    scn = bpy.context.scene
-    scn.name = name
-    scn.render.engine = 'CYCLES'
-    # scn.world.use_nodes = True
+    if not blender_scene:
+        bpy.ops.scene.new(type='NEW')
+        blender_scene = bpy.context.scene
 
-    # Always link in the whole node forest
-    scn.objects.link(op.armature_ob)
+    blender_scene.name = name
+    blender_scene.render.engine = 'CYCLES'
+    # blender_scene.world.use_nodes = True
 
+    # Always link in the whole node forest
+    # blender_scene.objects.link(op.armature_ob)
     roots = scene.get('nodes', [])
+
+    # Add a root object to fix the difference in orientation
+    root_object = bpy.data.objects.new("SceneRoot", None)
+    root_object.matrix_local = axis_conversion(from_forward="Z", from_up="Y").to_4x4()
+    blender_scene.objects.link(root_object)
+
     for root_idx in roots:
         # Link in any objects in this tree
-        for ob in op.root_to_objects[root_idx]:
-            scn.objects.link(ob)
+        for ob in create_node(op, root_idx):
+            ob.parent = root_object
+
+    return blender_scene
+
 
-    return scn
+def append_to_scene(op):
+    scene_idx = op.gltf.get('scene', 0)
+    op.scenes[scene_idx] = create_scene(op, scene_idx, bpy.context.scene)
```

Comment:
There aren't required to be any scenes in the glTF. This will raise an error if there aren't.

Reply:
Good point. Thanks for the feedback.